pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -286,7 +286,9 @@ in with passthru; stdenv.mkDerivation {
   CPPFLAGS = concatStringsSep " " (map (p: "-I${getDev p}/include") buildInputs);
   LDFLAGS = concatStringsSep " " (map (p: "-L${getLib p}/lib") buildInputs);
   LIBS = "${optionalString (!stdenv.isDarwin) "-lcrypt"} ${optionalString (ncurses != null) "-lncurses"}";
-  NIX_LDFLAGS = optionalString (stdenv.isLinux && !stdenv.hostPlatform.isMusl) "-lgcc_s" + optionalString stdenv.hostPlatform.isMusl "-lgcc_eh";
+  NIX_LDFLAGS =
+    optionalString (stdenv.isLinux && !stdenv.hostPlatform.isMusl && !(stdenv.hostPlatform.useLLVM or false)) "-lgcc_s"
+    + optionalString (stdenv.hostPlatform.isMusl && !(stdenv.hostPlatform.useLLVM or false)) "-lgcc_eh";
```

Comment:
I had

```nix
if stdenv.isLinux && stdenv.cc.isGNU then
  if stdenv.hostPlatform.isMusl then
    "-lgcc_eh"
  else
    "-lgcc_s"
else
  ""
```

I think it's more readable that way.

Reply:
fair enough, I just hacked this in. I think `stdenv.isLinux && stdenv.cc.isGNU` can be simplified to just `stdenv.cc.isGNU` in practice? Hard to tell whether there is actual merit to the `isLinux` thing or it was just a case of not darwin.

Reply:
I doubt it's necessary with e.g. gcc on Darwin.  So I think we should first check if cc is gcc, and then say if it's Musl do gcc_eh, and if it's Glibc do gcc_s.

Reply:
Done.